src/device-registry/utils/create-site.js

```diff
@@ -801,59 +794,52 @@ const manageSite = {
         requestForAirQloudsAndWeatherStations
       );
 
+      logObject("responseFromFindAirQlouds", responseFromFindAirQlouds);
       if (responseFromFindAirQlouds.success === true) {
         request["body"]["airqlouds"] = responseFromFindAirQlouds.data;
-      }
-
-      if (responseFromFindAirQlouds.success === false) {
+      } else if (responseFromFindAirQlouds.success === false) {
         logObject(
           "responseFromFindAirQlouds was unsuccessful",
           responseFromFindAirQlouds
         );
       }
 
-      const responseFromNearestWeatherStation = await manageSite.findNearestWeatherStation(
-        requestForAirQloudsAndWeatherStations
-      );
-
-      logObject(
-        "responseFromNearestWeatherStation",
-        responseFromNearestWeatherStation
-      );
-
-      if (responseFromNearestWeatherStation.success === true) {
-        let nearest_tahmo_station = responseFromNearestWeatherStation.data;
-        delete nearest_tahmo_station.elevation;
-        delete nearest_tahmo_station.countrycode;
-        delete nearest_tahmo_station.timezoneoffset;
-        delete nearest_tahmo_station.name;
-        delete nearest_tahmo_station.type;
-        request["body"]["nearest_tahmo_station"] = nearest_tahmo_station;
-      }
-
-      if (responseFromNearestWeatherStation.success === false) {
-        logObject(
-          "unable to find the nearest weather station",
-          responseFromNearestWeatherStation
-        );
-      }
+      // const responseFromNearestWeatherStation = await manageSite.findNearestWeatherStation(
```

Comment:
just commented out section for getting the nearest weather station. Shall reenable in future PR after refactoring.

Reply:
It was causing a 504 error of the sites refresh endpoint. There is a problem with the Tahmo weather API.

Reply:
Is it possible to set a request timeout Instead of commenting out the code @Baalmart

Reply:
Exactly!
That is what I will look into.... in a future PR as hinted in the comment section earlier. That is: "handling request time outs with the Axios client" @NoahNsimbe